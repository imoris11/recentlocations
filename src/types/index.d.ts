declare module 'redux-persist/lib/*';

export interface BaseAction {
  type: string;
  payload: any;
}

export interface User {
  id?: string;
  name: string;
  username: string;
  company: string;
}

export interface SingleItem {
  item: Location;
}

export interface ListItem {
  item: Location;
  removeListItem: Function;
}

export interface Location {
  formattedAddress?: string;
  timestamp: number;
  lat: number;
  lng: number;
}

export interface CurrentLocationType {
  formattedAddress?: string;
  timestamp?: string;
  username: string;
}
