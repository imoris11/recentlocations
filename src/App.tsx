import Home from './containers/Home';
import Map from './containers/Map';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import * as React from 'react';
import Icon from 'react-native-easy-icon';

const Tab = createBottomTabNavigator();

export type AppTabParamList = {
  Home: undefined;
  Settings: {userID?: string};
};

const App = () => {
  return (
    <Tab.Navigator
      screenOptions={({route}) => ({
        tabBarIcon: ({focused, color, size}) => {
          let iconName = '';

          if (route.name === 'Home') {
            iconName = focused ? 'home' : 'home-outline';
          } else if (route.name === 'Map') {
            iconName = focused ? 'map' : 'map-outline';
          }

          // You can return any component that you like here!
          return <Icon name={iconName} type="material-community" size={size} color={color} />;
        },
      })}>
      <Tab.Screen name="Home" component={Home} />
      <Tab.Screen name="Map" component={Map} />
    </Tab.Navigator>
  );
};

export default App;
