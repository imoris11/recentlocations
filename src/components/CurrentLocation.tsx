import React from 'react';
import {CurrentLocationType} from '../types';
import {View, Text, StyleSheet} from 'react-native';

export const CurrentLocation = ({username, formattedAddress, timestamp}: CurrentLocationType) => {
  const nameArray = username?.split(' ');
  const abbreviatedName = nameArray && nameArray[0][0] + nameArray[1][0];
  return (
    <View style={styles.currentLocationContainer}>
      <Text style={styles.sectionHeader}>Current Location</Text>
      <View style={styles.currentLocationBody}>
        <View style={styles.avatarContainer}>
          <Text style={styles.avatarText}>{abbreviatedName}</Text>
        </View>
        <View style={styles.location}>
          <Text numberOfLines={1} style={styles.locationText}>
            {formattedAddress}
          </Text>
          <Text style={styles.timeStamp}>{timestamp}</Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  currentLocationContainer: {
    marginVertical: 10,
  },

  currentLocationBody: {
    flexDirection: 'row',
    marginVertical: 15,
    alignItems: 'center',
  },
  avatarContainer: {
    height: 40,
    width: 40,
    borderRadius: 20,
    backgroundColor: '#ee6723',
    justifyContent: 'center',
    alignItems: 'center',
  },
  avatarText: {
    color: '#FFF',
    fontWeight: '800',
  },
  location: {
    marginLeft: 10,
    marginRight: 40,
  },
  locationText: {
    fontWeight: '500',
    fontSize: 16,
  },
  timeStamp: {
    color: 'grey',
    marginTop: 2,
  },
  sectionHeader: {
    color: 'grey',
    fontWeight: '600',
  },
});
