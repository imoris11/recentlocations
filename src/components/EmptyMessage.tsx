import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

export const EmptyMessage = () => {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>No locations</Text>
      <Text style={styles.text}>Activate location to see current locations</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontSize: 18,
    fontWeight: '600',
    textAlign: 'center',
  },
});
