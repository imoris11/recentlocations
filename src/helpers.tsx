export const convertTimestampToDate = (timestamp: number) => {
  if (!timestamp) return;
  const date = new Date(timestamp * 1000);
  const day = date.getDate();
  const month = date.getMonth() + 1;
  const year = date.getFullYear();
  const hours = date.getHours();
  const minutes = '0' + date.getMinutes();
  const seconds = '0' + date.getSeconds();
  const formattedTime = `${day}/${month}/${year} ${hours}:${minutes.substr(-2)}:${seconds.substr(
    -2,
  )}`;
  return formattedTime;
};
