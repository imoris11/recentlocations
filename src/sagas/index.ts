import {all} from 'redux-saga/effects';
import users from './usersSagas';
import location from './locationSagas';

export default function* root() {
  yield all([users(), location()]);
}
