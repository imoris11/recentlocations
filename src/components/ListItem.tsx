import React from 'react';
import {Text, View, TouchableOpacity, StyleSheet} from 'react-native';
import {ListItem} from 'src/types';
import {convertTimestampToDate} from '../helpers';

export const SingleListItem = ({item, removeListItem}: ListItem) => {
  return (
    <View style={styles.container} key={item.timestamp}>
      <View style={styles.bodyContainer}>
        <Text numberOfLines={1} style={styles.locationText}>
          {item.formattedAddress}
        </Text>
        <Text style={styles.timeStamp}>{convertTimestampToDate(item.timestamp)}</Text>
      </View>
      <View style={styles.buttonContainer}>
        <TouchableOpacity onPress={() => removeListItem(item.timestamp)} style={styles.button}>
          <Text>Remove</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default SingleListItem;

const styles = StyleSheet.create({
  bodyContainer: {
    flex: 1,
  },
  location: {
    marginLeft: 10,
  },
  locationText: {
    fontWeight: '500',
    fontSize: 16,
  },
  timeStamp: {
    color: 'grey',
    marginTop: 2,
  },
  sectionHeader: {
    color: 'grey',
    fontWeight: '600',
  },
  buttonContainer: {
    justifyContent: 'flex-end',
  },
  button: {
    backgroundColor: 'lightgrey',
    justifyContent: 'center',
    borderRadius: 3,
    paddingHorizontal: 10,
    height: 30,
  },
  container: {
    marginTop: 20,
    flexDirection: 'row',
  },
});
