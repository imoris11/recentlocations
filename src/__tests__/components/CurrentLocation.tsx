import React from 'react';
import {render} from 'react-native-testing-library';
import {CurrentLocation} from '../../components/CurrentLocation';
import {CurrentLocationType} from 'src/types';

describe('CurrentLocation', () => {
  const mockData = {} as CurrentLocationType;
  const renderCurrentLocation = (props = {}) =>
    render(<CurrentLocation {...mockData} {...props} />);
  it('renders correctly', () => {
    const {toJSON} = renderCurrentLocation();
    expect(toJSON).toMatchSnapshot();
  });

  it('accurately abbreviates the name', () => {
    const {getByText} = renderCurrentLocation({username: 'Richard Igbiriki'});
    const element = getByText('RI');
    expect(element.children).toHaveLength(1);
  });

  it('returns 0 wrong abbreviations', () => {
    const {queryAllByText} = renderCurrentLocation({username: 'Richard Igbiriki'});
    const element = queryAllByText('RIG');
    expect(element.length).toBe(0);
  });

  it('shows the right formattedAddress', () => {
    const {queryAllByText} = renderCurrentLocation({formattedAddress: 'Bayelsa Nigeria'});
    const element = queryAllByText('Bayelsa Nigeria');
    expect(element.length).toEqual(1);
  });
});
