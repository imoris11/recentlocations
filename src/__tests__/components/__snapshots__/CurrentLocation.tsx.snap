// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentLocation renders correctly 1`] = `[Function]`;
